packages/cdktf-cli/lib/synth-stack.ts

```diff
@@ -49,11 +49,18 @@ export class SynthStack {
       path.join(outdir, Manifest.stacksFolder)
     );
 
+    const env = Object.entries(process.env)
+      .filter(
+        // We don't want to pass Terraform variables to the synth command since they should only be used at execution time
+        ([key]) => !key.startsWith("TF_VAR_")
+      )
+      .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
```

Comment:
You could also use `Object.fromEntries` instead of `.reduce()` as its available since Node v12. I think it might make it more readable.